web/client/epics/geostory.js

```diff
@@ -166,20 +215,26 @@ export const loadGeostoryEpic = (action$, {getState = () => {}}) => action$
         .switchMap( ({id}) =>
             Observable.defer(() => {
                 if (id && isNaN(parseInt(id, 10))) {
-                    return axios.get(`configs/${id}.json`);
+                    return axios.get(`configs/${id}.json`)
+                        // not return anything else that data in this case
+                        // to match with data/resource object structure of getResource
+                        .then(({data}) => ({data}));
                 }
-                // TODO manage load process from external api
-                return axios.get(`configs/sampleStory.json`);
+                return getResource(id);
             })
-            .map(({ data }) => {
-                if (isObject(data)) {
-                    return setCurrentStory(data);
+            .filter(({data}) => {
```

Comment:
Can you explain why the filter if you throw the error?  to avoid to launch following actions, but the throw should "jump" the flow tot the catch method, right?

Reply:
Yes, maybe a `do` method was more appropriate for this. I wanted a method that didn't affect the result, just for check.
I'll replace it with do
```suggestion
            .do(({data}) => {
```